### PR TITLE
WID-174. TS widget with plotly

### PR DESCRIPTION
**Pull request by romina1601**
Still a work in progress. What needs to be done:

- [x] Create a base class for both TS widgets
- [x] Checkboxes for channel selection and redraw the plot, not just hide the traces when deselecting a signal
- [x] Update notebooks to use and demonstrate the new API
- [x] Create multiple modules from the ts_widget.py file
- [x] Update unit tests

(Maybe) more points to be added.

### CODE REVIEW

File: tvbwidgets/ui/ts_widget.py [hunk 1]
```diff
@@ -545,6 +587,210 @@ def _update_fig(self):
...
+
+        # get data and names for selected channels; self.raw is updated before redrawing starts
+        data, _ = self.raw[:, :]
+        data = data[picks, :]
+        ch_names = [ch_names[i] for i in picks]
+
+        # redraw the entire plot
+        self.plot_ts_with_plotly(data, ch_names)
+
+
```
Comment (liadomide) [inline tvbwidgets/ui/ts_widget.py:793]: This file has become large (to read/manipulate). 
Would it be worthy to split it in 2 or 3 (maybe in its own `ts` folder?

Comment (romina1601) [reply tvbwidgets/ui/ts_widget.py:793]: Yup, I agree. I will create a dedicated TS folder for the TS widget and break this file into multiple modules.

**Comment by liadomide** [comment]
I think a rebase with master is necessary here.
Also Sonar has 7 code smells that at least partially could be handled before this is merged

**Review by liadomide — APPROVED** [review]
It looks ok as code. We could merge.
Although the overall UI in the Plotly viewer might be polished a bit more (e.g. when expanding the channels the left positioned labels change their aspect, as if the overall elem is changing size, or the bottom area seems somehow unfinished - not aligned with the above parts ...).

**Pull request merged**